### ISSUE TITLE
【论文复现赛】nn.BatchNorm2D 参数数量和torch nn.BatchNorm2d不同

### ISSUE THREAD

**Issue by justld** [PaddlePaddle/Paddle#37852]
paddle环境：Aistudio  gpu至尊，paddlepaddle==2.2.0
paddle代码如下：

      import paddle
      from paddle import nn
      
      model = nn.Sequential(
          nn.BatchNorm2D(64),
          nn.Conv2D(64, 3, 1),
      )
      paddle.summary(model, (4, 64, 256, 256))

打印结果如下：

        ---------------------------------------------------------------------------
         Layer (type)       Input Shape          Output Shape         Param #    
        ===========================================================================
        BatchNorm2D-11  [[4, 64, 256, 256]]   [4, 64, 256, 256]         256      
           Conv2D-5     [[4, 64, 256, 256]]    [4, 3, 256, 256]         195      
        ===========================================================================
        Total params: 451
        Trainable params: 195
        Non-trainable params: 256
        ---------------------------------------------------------------------------
        Input size (MB): 64.00
        Forward/backward pass size (MB): 134.00
        Params size (MB): 0.00
        Estimated Total Size (MB): 198.00
        ---------------------------------------------------------------------------
有2个问题：
1、BatchNorm2D有参数γ和β可学习，输入通道64，参数数量应该是2 * 64 = 128，这里是256；
2、BatchNorm2D参数是 non-trainable的；  

torch版本：1.8.1 + cuda 10.2
代码如下：
        
        from torchsummary import summary
        
        import torch
        from torch import nn
        
        model = nn.Sequential(
            nn.BatchNorm2d(64),
            nn.Conv2d(64, 3, 1, bias=False)
        )
        
        print(torch.__version__)
        summary(model, input_size=(64, 256, 256), batch_size=4)

输出结果如下：
        
        ----------------------------------------------------------------
                Layer (type)               Output Shape         Param #
        ================================================================
               BatchNorm2d-1          [4, 64, 256, 256]             128
                    Conv2d-2           [4, 3, 256, 256]             192
        ================================================================
        Total params: 320
        Trainable params: 320
        Non-trainable params: 0
        ----------------------------------------------------------------
        Input size (MB): 64.00
        Forward/backward pass size (MB): 134.00
        Params size (MB): 0.00
        Estimated Total Size (MB): 198.00
        ----------------------------------------------------------------
torch的BatchNorm2d参数数量正确，且可以训练。

**Comment by wangxicoding** [comment]
![image](https://user-images.githubusercontent.com/10208305/144746718-bc17e52f-1fcf-4791-9b61-393801e11b96.png)
1、Paddle里面应该是把mean和variance也给统计进去了，所以是4*64=128。torch应该是用buffer实现的mean和variance
2、直接使用`.trainable`查看weight和bias是可训练的，mean和variance是non-trainable的。 这个可能是`paddle.summary`什么地方有问题，还得再定位一下

**Comment by justld** [comment]
感谢解答

**Comment by Ligoml** [comment]
![image](https://user-images.githubusercontent.com/39876205/144816787-85763db8-de28-41ad-867c-b3ef1e60a53d.png)
关于 torch 模型转换为 paddle 的细节问题，在论文复现指南中有详细的介绍，欢迎参考文档开展论文复现工作：
https://github.com/PaddlePaddle/models/blob/tipc/docs/lwfx/ArticleReproduction_CV.md#42-%E6%A8%A1%E5%9E%8B%E7%BB%93%E6%9E%84%E5%AF%B9%E9%BD%90

**Comment by Ligoml** [comment]
相关参考资料：
论文复现指南：https://github.com/PaddlePaddle/models/blob/tipc/docs/lwfx/ArticleReproduction_CV.md
视频教程：https://aistudio.baidu.com/aistudio/education/group/info/24681
